packages/amplication-client/src/Entity/RelationFkHolderField.tsx

```diff
@@ -0,0 +1,80 @@
+import { SelectField, SelectFieldProps } from "@amplication/design-system";
+import { useQuery } from "@apollo/client";
+import { useFormikContext } from "formik";
+import { useEffect, useRef } from "react";
+import * as models from "../models";
+import { GET_ENTITY_FIELD_BY_PERMANENT_ID } from "./RelatedEntityFieldField";
+
+type Props = Omit<SelectFieldProps, "options"> & {
+  entity: models.Entity;
+};
+
+const RelationFkHolderField = ({ entity, ...props }: Props) => {
+  const formik = useFormikContext<models.EntityField>();
+
+  const entityFieldRef: React.MutableRefObject<{
+    relatedField: models.EntityField | undefined;
+    isOneToOne: boolean;
+    entityListOptions: { value: string; label: string }[];
+  }> = useRef({
+    relatedField: undefined,
+    isOneToOne: false,
+    entityListOptions: [],
+  });
+  const { data } = useQuery<{ entity: models.Entity }>(
+    GET_ENTITY_FIELD_BY_PERMANENT_ID,
+    {
+      variables: {
+        entityId: formik.values.properties.relatedEntityId,
+        fieldPermanentId: formik.values.properties.relatedFieldId,
+      },
+      skip:
+        !formik.values.properties?.relatedEntityId ||
+        !formik.values.properties?.relatedFieldId,
+    }
+  );
+
+  useEffect(() => {
```

Comment:
I understand what you did here, but I think the useMemo version is much more readable and straight-forward - don't see the advantage of using the ref object. Also, useMemo is consistent with the way we do it across the client app
I don't ask you to change, but I just want to have a discussion about the best practice